### PR TITLE
Implement binary to decimal

### PR DESCRIPTION
**Pull request by neha-saggam**
# Welcome to JavaScript community

### **Describe your change:**

* Implemented binary to decimal

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new JavaScript files are placed inside an existing directory.
* [x] All filenames should use the UpperCamelCase (PascalCase) style.  There should be no spaces in filenames.
     **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by itsvinayak** [comment]
@neha-saggam please read  [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md) some of the tests are falling
thank you
@itsvinayak

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by neha-saggam** [comment]
@itsvinayak When I run `npm run test` it fails on local saying `echo "Error: no test specified" && exit 1` how do I resolve it ? Do I have to write tests for this, if yes I don't see any tests for reference

**Comment by itsvinayak** [comment]
currently, this repo does not contain any test as such, 
run `npx standard --fix`   [read more here](https://www.npmjs.com/package/standard) to format the code

**Comment by neha-saggam** [comment]
@itsvinayak Thank you! All checks have passed now 🙂 , it was missing end of line. I had already ran the npm standards command but somehow missed this. Maybe we can add a pre-push hook for this ? Thoughts ?

File: Conversions/BinaryToDecimal.js [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-    console.log(`Decimal of ${binaryNumber} is ${decimalNumber}`)
-  }
-  
-  binaryToDeicmal('111001')
-  binaryToDeicmal('101')
-  
\ No newline at end of file
+  let decimalNumber = 0
+  const binaryDigits = binaryNumber.split('').reverse()
+  binaryDigits.forEach((binaryDigit, index) => {
```
Comment (itsvinayak) [inline Conversions/BinaryToDecimal.js:4]: @neha-saggam please add some hints and comment for better understanding
thank you,
@itsvinayak

**Review by itsvinayak — CHANGES_REQUESTED** [review]

**Comment by itsvinayak** [comment]
> @itsvinayak Thank you! All checks have passed now  , it was missing end of line. I had already ran the npm standards command but somehow missed this. Maybe we can add a pre-push hook for this ? Thoughts ?

we are working on pre-push hook, you are welcome to implement ...

File: Conversions/BinaryToDecimal.js (hunk 1 above)
Comment (neha-saggam) [reply Conversions/BinaryToDecimal.js]: @itsvinayak I personally feel the code should be enough self-explanatory. Comments are considered as anti-patterns.
I have still added some comments but not really convinced of why do we need it. Please let me know if I am missing something here. :)

**Pull request merged**